res/skins/QMLDemo/WaveformShader.qml

```diff
@@ -0,0 +1,222 @@
+import Mixxx 0.1 as Mixxx
+import QtQuick 2.12
+
+ShaderEffect {
+    id: root
+
+    property string group // required
+    property var deckPlayer: Mixxx.PlayerManager.getPlayer(group)
+    property variant framebufferSize: Qt.size(width, height)
+    property int waveformLength: root.deckPlayer.waveformLength
+    property int textureSize: root.deckPlayer.waveformTextureSize
+    property int textureStride: root.deckPlayer.waveformTextureStride
+    property real firstVisualIndex: 1
+    property real lastVisualIndex: root.deckPlayer.waveformLength / 2
+    property color axesColor: "#FFFFFF"
+    property color highColor: "#FF0000"
+    property color midColor: "#00FF00"
+    property color lowColor: "#0000FF"
+    property real highGain: filterWaveformEnableControl.value ? (filterHighKillControl.value ? 0 : filterHighControl.value) : 1
+    property real midGain: filterWaveformEnableControl.value ? (filterMidKillControl.value ? 0 : filterMidControl.value) : 1
+    property real lowGain: filterWaveformEnableControl.value ? (filterLowKillControl.value ? 0 : filterLowControl.value) : 1
+    property real allGain: pregainControl.value
+    property variant waveformTexture
+
+    fragmentShader: "
```

Comment:
not sure if you changed this much from what there already is in `res/shaders` but if you didn't, you can use the file path instructions things in qt. like `qrc://` and just plain `file:` (for referring to a relative path).

Reply:
The only change is s/gl_TexCoord[0]/qt_TexCoord0/.

Reply:
Ok, I just mean that I dislike the inline glsl string and rather have it in a separate file. You could've just linked to the one in `res/shaders` if there weren't any changes.

Reply:
I know, but this doesn't work unfortunately. This PR is basically unmergeable anyway because the cpp part is so hacky. I just opened it so that someone has a starting part in case he/she wants to work on scrolling QML waveforms.